### PR TITLE
[stable/elasticsearch-exporter] Let the chart have the ability to export the ports in the nodeport mode (#1…

### PR DESCRIPTION
**Pull request by jinleileiking**
Let the chart have the ability to export the ports in the nodeport mode

### CODE REVIEW

**Comment by jinleileiking** [comment]
****

**Comment by jinleileiking** [comment]
What should I do :(

**Comment by jinleileiking** [comment]
@desaintmartin @svenmueller @caarlos0   Any ideas?

**Comment by desaintmartin** [comment]
Thanks for the contribution!
Could you:
- Describe the change in the PR description
- Bump minor version instead of patch since this is new feature
- Document in README.md the new values
- Make them optional (commented-out in values.yaml by default, and template should check for its existence before applying it)
?

**Comment by jinleileiking** [comment]
@desaintmartin @caarlos0 all done. Thanks for another review :)

I would like to be a maintainer for this project too.

File: stable/elasticsearch-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Elasticsearch stats exporter for Prometheus
 name: elasticsearch-exporter
-version: 2.1.1
+version: 2.2.1
```
Comment (caarlos0) [inline stable/elasticsearch-exporter/Chart.yaml]: should be 2.2.0

File: stable/elasticsearch-exporter/templates/service.yaml [hunk 2]
```diff
@@ -19,6 +19,10 @@ spec:
   ports:
     - name: {{ .Values.service.metricsPort.name }}
       port: {{ .Values.service.httpPort }}
+{{- if eq .Values.service.type "NodePort"}}
```
Comment (caarlos0) [inline stable/elasticsearch-exporter/templates/service.yaml]: imho this and the `end` should be in the same indent level as `port` above

Comment (jinleileiking) [reply stable/elasticsearch-exporter/templates/service.yaml]: done

**Comment by jinleileiking** [comment]
@caarlos0 Please take your time for another review. Thanks. Sorry for the interruption.

File: stable/elasticsearch-exporter/README.md [hunk 3]
```diff
@@ -66,8 +66,10 @@ Parameter | Description | Default
 `podSecurityPolicies.enabled` | Enable/disable PodSecurityPolicy and associated Role/Rolebinding creation | `false`
 `serviceAccount.create` | Create a ServiceAccount for the pod | `false`
 `serviceAccount.name` | Name of a ServiceAccount to use that is not handled by this chart | `default`
-`service.type` | type of service to create | `ClusterIP`
+`service.type` | type of service to create, `ClusterIP` or `NodePort` | `ClusterIP`
```
Comment (desaintmartin) [inline stable/elasticsearch-exporter/README.md:69]: There are other valid types.

File: stable/elasticsearch-exporter/README.md [hunk 4]
```diff
@@ -66,8 +66,10 @@ Parameter | Description | Default
...
 `service.httpPort` | port for the http service | `9108`
+`service.targetPort` | port for the http service that the pod exported | `9108`
```
Comment (desaintmartin) [inline stable/elasticsearch-exporter/README.md:71]:
```suggestion
`service.targetPort` | port for the http service that the pod exported | `nil`
`service.nodePort` | port for the http service that node exported in NodePort mode | `nil`
```

File: stable/elasticsearch-exporter/templates/service.yaml [hunk 5]
```diff
@@ -19,6 +19,10 @@ spec:
   ports:
     - name: {{ .Values.service.metricsPort.name }}
       port: {{ .Values.service.httpPort }}
+      {{- if eq .Values.service.type "NodePort"}}
```
Comment (desaintmartin) [inline stable/elasticsearch-exporter/templates/service.yaml:22]: nodePort can be valid for LoadBalancer type too. Maybe it is better not to have too much intelligence here and just have two conditions (if targetPort, then set it, if nodePort, then set it).

File: stable/elasticsearch-exporter/templates/service.yaml [hunk 6]
```diff
@@ -19,6 +19,10 @@ spec:
...
+      targetPort: {{ .Values.service.targetPort }}
```
Comment (desaintmartin) [inline stable/elasticsearch-exporter/templates/service.yaml:23]: I'm curious: what is the use case where the port is different than the targetPort?

Note that kube doc reads `If this is not specified, the value of the 'port' field is used (an identity map)`

**Review by desaintmartin — CHANGES_REQUESTED** [review]

File: stable/elasticsearch-exporter/README.md (hunk 3 above)
Comment (jinleileiking) [reply stable/elasticsearch-exporter/README.md:69]: should I and loadbalancer and ingress?

Comment (caarlos0) [reply stable/elasticsearch-exporter/README.md:69]: I would just leave it as `type of service to create`

But FWIW take a look on other charts and see how they do...

**Review by svenmueller — COMMENTED** [review]
i don't see the need of having to define `spec.ports[*].nodePort`

**Comment by jinleileiking** [comment]
No needs. Ingress is ok.

**Pull request closed without merging**